com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipelineGlobalSettings.cs

```diff
@@ -0,0 +1,794 @@
+        {
+            if (AreEditorResourcesCreated())
+                return;
+
+            var editorResourcesPath = HDUtils.GetHDRenderPipelinePath() + "Editor/RenderPipelineResources/HDRenderPipelineEditorResources.asset";
+            var objs = InternalEditorUtility.LoadSerializedFileAndForget(editorResourcesPath);
+            m_RenderPipelineEditorResources = (objs != null && objs.Length > 0) ? objs[0] as HDRenderPipelineEditorResources : null;
+            if (forceReload)
+            {
+                if (ResourceReloader.ReloadAllNullIn(m_RenderPipelineEditorResources, HDUtils.GetHDRenderPipelinePath()))
+                {
+                    InternalEditorUtility.SaveToSerializedFileAndForget(
+                        new Object[] { m_RenderPipelineEditorResources },
+                        editorResourcesPath,
+                        true);
+                }
+            }
+            else if (!EditorUtility.IsPersistent(m_RenderPipelineEditorResources))
+            {
+                m_RenderPipelineEditorResources = AssetDatabase.LoadAssetAtPath<HDRenderPipelineEditorResources>(editorResourcesPath);
+            }
+        }
+
+        internal bool AreEditorResourcesCreated()
+        {
+            return (m_RenderPipelineEditorResources != null && !m_RenderPipelineEditorResources.Equals(null));
+        }
+
+        // Note: This function is HD specific
+        /// <summary>HDRP default Decal material.</summary>
+        public Material GetDefaultDecalMaterial()
+            => m_RenderPipelineEditorResources.materials.defaultDecalMat;
+
+        // Note: This function is HD specific
+        /// <summary>HDRP default mirror material.</summary>
+        public Material GetDefaultMirrorMaterial()
+            => m_RenderPipelineEditorResources.materials.defaultMirrorMat;
+#endif
+
+        #endregion //Editor Resources
+
+        #region Ray Tracing Resources
+        [SerializeField]
+        HDRenderPipelineRayTracingResources m_RenderPipelineRayTracingResources;
+        internal HDRenderPipelineRayTracingResources renderPipelineRayTracingResources
+        {
+            get { return m_RenderPipelineRayTracingResources; }
+            set { m_RenderPipelineRayTracingResources = value; }
+        }
+
+#if UNITY_EDITOR
+        internal void EnsureRayTracingResources(bool forceReload)
+        {
+            if (AreRayTracingResourcesCreated())
+                return;
+
+            m_RenderPipelineRayTracingResources = UnityEditor.AssetDatabase.LoadAssetAtPath<HDRenderPipelineRayTracingResources>(HDUtils.GetHDRenderPipelinePath() + "Runtime/RenderPipelineResources/HDRenderPipelineRayTracingResources.asset");
+
+            if (forceReload)
+            {
+#if UNITY_EDITOR_LINUX // Temp hack to be able to make linux test run. To clarify
+                ResourceReloader.TryReloadAllNullIn(m_RenderPipelineRayTracingResources, HDUtils.GetHDRenderPipelinePath());
```

Comment:
This seams dangerous as we potentially hide error at import time (beside asset data base not ready). @sebastienlagarde @JulienIgnace-Unity  for awarness (it is already as this in current code). This function was originally done for Universal that use lazy init. We need full init in HDRP as we do not attempt to load things again once pipeline fully created.

Reply:
We addressed this in a call with Thomas Chollet, i guess we would still need to know why it was necessary. @JulienIgnace-Unity and @sebastienlagarde do you remember?

Reply:
Just realized I did not answer this one: I have no idea why this workaround was need though :p